### ISSUE TITLE
feat(weapons): implement autonomous ZMQ inference worker for E2E telemetry

### ISSUE THREAD

**Issue by mauriziokeningen** [mauriziokeningen/Computer-Vision-Residential-Security-System#35]
# feat(weapons): implement autonomous ZMQ inference worker for E2E telemetry

## Executive Summary & Context
To achieve End-to-End (E2E) system validation, we must deploy the Weapon Detection microservice (`src/modules/weapons/inference.py`). We are adopting a "Tracer Bullet" development methodology: our immediate priority is establishing a robust, fault-tolerant Inter-Process Communication (IPC) pipeline from the edge-node camera to the Orchestrator sink. 

Furthermore, to support non-destructive, vector-based rendering on the future React Frontend, this worker must extract and transmit absolute Bounding Box coordinates. This architectural decoupling ensures the UI can draw dynamic DOM overlays without permanently mutating or burning pixels into the raw evidence frames.

## Architectural Directives & Network Contracts

The worker must act as an autonomous node in the Fan-Out topology and adhere strictly to the following constraints:

**1. Memory Safety & Throttling (Chesterton's Fence):**
The worker must subscribe to the video stream via `zmq.SUB`. Crucially, the socket MUST be configured with `socket.setsockopt(zmq.CONFLATE, 1)`. If inference time exceeds the 10 FPS ingestion rate, the worker must explicitly drop stale frames rather than queuing them. Failure to implement this will result in fatal out-of-memory (OOM) operating system kills.

**2. Cold-Start Mitigation:**
PyTorch/CUDA environments suffer from high latency during the initial tensor allocation. The worker must execute a "warmup" pass using a dummy zero-tensor during bootstrapping to prevent processing delays when the first real threat is captured.

**3. IPC Payload Contract (Updated for Vector UI):**
The worker must exclusively push (`zmq.PUSH`) data to the Orchestrator (`tcp://127.0.0.1:5556`) when a threat exceeds the confidence threshold (>= 0.50). Silent frames should generate zero network traffic. The payload must include the `bbox` array `[x_min, y_min, x_max, y_max]` cast as integers.

```json
{
  "module": "weapons",
  "camera_id": "main_camera",
  "timestamp": "2026-04-12T15:00:00.123456",
  "frame_data": "<base64_encoded_jpeg>", 
  "detections": [
    {
      "class": "pistol", 
      "confidence": 0.88,
      "bbox": [150, 200, 350, 400]
    }
  ]
}
```

## Implementation Steps
1. Populate `src/modules/weapons/inference.py` with the approved ZMQ worker boilerplate.
2. Extract the spatial coordinates from the YOLO tensor (`box.xyxy[0].cpu().numpy().astype(int).tolist()`) and append them to the detection dictionary to satisfy the UI contract.
3. Link the `MODEL_WEIGHTS` constant to the absolute path of the currently available `best.pt` file generated from the initial Roboflow experiments.
4. Integrate the worker's execution entry point into the `processes` array within `main.py`.

## Acknowledged Technical Debt
The current model weights integrated via this issue are baseline iterations and possess known performance degradation regarding small/distant objects. This is documented and accepted. A subsequent Epic will address SOTA Fine-Tuning using CCTV-specific datasets. Once that training is complete, the new weights will be hot-swapped into this pipeline without requiring code modifications.

## Architectural Decision: YAGNI & Stateless Domain
Reviewers note: Unlike the `face` module (which is stateful and requires database lookups/embeddings isolated in `services/face_processor.py`), weapon detection is a pure, stateless I/O function. Applying YAGNI (You Aren't Gonna Need It), we intentionally bypassed creating a redundant `services/weapon_processor.py`. All parsing logic resides within the edge worker to prevent over-engineering and eliminate unnecessary call-stack latency.

## Definition of Done (DoD)
- [ ] The weapon worker process boots successfully and reserves its VRAM allocation without crashing.
- [ ] The worker accurately drops frames (via `zmq.CONFLATE`) when intentionally throttled, demonstrating immunity to buffer bloat.
- [ ] Spatial coordinates (`bbox`) are successfully extracted from the PyTorch tensor and formatted as an integer array.
- [ ] Presenting a weapon to the camera successfully generates the strict JSON payload (including `bbox`) and transmits it to the Orchestrator.
